packages/graphql/src/classes/utils/asserts-indexes-and-constraints.ts

```diff
@@ -172,63 +202,43 @@ async function checkIndexesAndConstraints({ nodes, session }: { nodes: Node[]; s
 
     debug("Successfully checked for the existence of all necessary constraints");
 
-    const existingIndexes: Record<string, { labelsOrTypes: string; properties: string[] }> = {};
+    const existingIndexes = await getExistingIndexes({ session });
     const indexErrors: string[] = [];
-    const indexesCypher = "SHOW INDEXES";
-
-    debug(`About to execute Cypher: ${indexesCypher}`);
-    const indexesResult = await session.run(indexesCypher);
-
-    indexesResult.records.forEach((record) => {
-        const index = record.toObject();
-
-        if (index.type !== "FULLTEXT" || index.entityType !== "NODE") {
-            return;
-        }
 
-        if (existingIndexes[index.name]) {
-            return;
-        }
-
-        existingIndexes[index.name] = {
-            labelsOrTypes: index.labelsOrTypes,
-            properties: index.properties,
-        };
-    });
-
-    nodes.forEach((node) => {
-        if (node.fulltextDirective) {
-            node.fulltextDirective.indexes.forEach((index) => {
-                // TODO: remove indexName assignment and undefined check once the name argument has been removed.
-                const indexName = index.indexName || index.name;
+    for (const entity of schemaModel.concreteEntities) {
+        if (entity.annotations.fulltext) {
+            entity.annotations.fulltext.indexes.forEach((index) => {
+                const indexName = index.indexName || index.name; // TODO remove indexName assignment and undefined check once the name argument has been removed.
                 if (indexName === undefined) {
                     throw new Error("The name of the fulltext index should be defined using the indexName argument.");
                 }
                 const existingIndex = existingIndexes[indexName];
                 if (!existingIndex) {
-                    indexErrors.push(`Missing @fulltext index '${indexName}' on Node '${node.name}'`);
+                    indexErrors.push(`Missing @fulltext index '${indexName}' on Node '${entity.name}'`);
 
                     return;
                 }
 
                 index.fields.forEach((field) => {
-                    const stringField = node.primitiveFields.find((f) => f.fieldName === field);
-                    const fieldName = stringField?.dbPropertyNameUnescaped || field;
+                    const attributeAdapter = entity.findAttribute(field);
```

Comment:
not an adapter

Reply:
also it feels like this code is extremely similar to the else above in createIndexesAndConstraints. Is there a benefit to extracting this into a utility function?